ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -134,10 +138,16 @@ private ClientImpl(final ClientOptions clientOptions, final Vertx vertx,
           KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR,
           serializedConsistencyVector.get());
     }
+    if (PushOffsetVector.isContinuationTokenEnabled(properties)) {
+      requestProperties.put(
```

Comment:
we supply the saved continuation token here when we do a retry

Reply:
We also need to set `KSQL_QUERY_PUSH_V2_CONTINUATION_TOKENS_ENABLED` and `KSQL_QUERY_PUSH_V2_ENABLED` for the feature to work. We should probably set `KSQL_QUERY_PUSH_V2_CONTINUATION_TOKENS_ENABLED` since it's a bit internal, if `PushOffsetVector.isContinuationTokenEnabled(properties)` is true (regardless of if we have a continuation token yet).  Currently, you just set it in the test